app/controllers/users_controller.rb

```diff
@@ -0,0 +1,70 @@
+class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user,   only: [:show, :edit, :update]
+  before_action :admin_user,     only: :destroy
+
+  def show
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      log_in @user
+      flash[:success] = "Welcome to the Eureka Caffee!"
+      redirect_to @user
+    else
+      render 'new'
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update_attributes(user_params)
+      flash[:success] = "Profile updated"
+      redirect_to @user
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    User.find(params[:id]).destroy
```

Comment:
This means I can destroy any user? Or is this controller for admins only? If yes, then create an admin namespace

Reply:
For action :destroy, there is a before_action :admin_user that prevents you from destroying any user.

Reply:
Ok, so this is an admin only action. In this case, please create and Admin namespace and add a controller there.